source/extensions/common/dynamic_forward_proxy/dns_cache.h

```diff
@@ -104,12 +104,26 @@ class DnsCache {
    *             target ports.
    * @param default_port supplies the port to use if the host does not have a port embedded in it.
    * @param callbacks supplies the cache load callbacks to invoke if async processing is needed.
-   * @return a cache load handle, in which case the callbacks will be invoked at a later time,
-   *         or nullptr if the cache is already loaded. In this case, callbacks will never be
-   *         called.
+   * @return a cache load result which includes both a status and handle. If the handle is non-null
+   *         the callbacks will be invoked at a later time, otherwise consult the status for the
+   *         reason the cache is not loading. In this case, callbacks will never be called.
    */
-  virtual LoadDnsCacheHandlePtr loadDnsCache(absl::string_view host, uint16_t default_port,
-                                             LoadDnsCacheCallbacks& callbacks) PURE;
+  enum class LoadDnsCacheStatus {
+    // The cache is already loaded. Callbacks will not be called.
```

Comment:
cache -> cache entry here and below?
I think in API naming there may be conflation of the cache and a cache entry, but could be we just don't overload the terms here and it's commonly done?

Reply:
This reads OK to me, but no objections to change it, at least in comments. I will take a pass through.